app/main.py

```diff
@@ -1 +1,35 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive: list[Animal] = []
+
+    def __init__(
+        self, name: str, health: int = 100, hidden: bool = False
+    ) -> None:
+        self.name: str = name
+        self.health: int = health
+        self.hidden: bool = hidden
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
```

Comment:
Use better = True, because hide shouldn't unhide animal

Reply:
I agree, but the tests explicitly check if this method changes value to the opposite